core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -215,14 +227,110 @@ public boolean isAvailable() {
       return file.exists();
    }
 
+   private boolean isCorruptData() throws IOException {
+      ByteBuffer buf = ByteBuffer.allocate(KEY_POS_LATEST);
+      long pos = filePos;
+      buf = readChannel(buf, pos, KEY_POS_LATEST);
+      if (buf.remaining() > 0)
+         return false;
+      buf.flip();
+      FileEntry fe = new FileEntry(pos, buf);
+      // Explicitly calculate the correct size as incorrect value stored in fe.size
+      int actualSize = KEY_POS_LATEST + fe.keyLen + fe.dataLen + fe.metadataLen + fe.internalMetadataLen;
+
+      // Attempt to read the next entry in the file
+      pos += actualSize;
+      buf = readChannel(buf, pos, KEY_POS_LATEST);
+
+      // Only single FileEntry present with correct size, so data is not corrupt
```

Comment:
actually, where are we checking if the first entry is not corrupted?

Reply:
The first entry cannot be corrupted. (The internal metatadata an expiry time can be lost)

There are two issues that the migration caused.

1. The expiryTime and internal metadata were completely lost on ALL entries. We cannot recover from this and is what it is.
2. There is a bunch of bytes between each entry that causes the file format to be corrupted. This is what skipping the bytes fixes. So we read the next entry at the correct location.

So this just checks for #2 really.